WikiDrips/WikiTableViewController.swift

```diff
@@ -58,31 +72,47 @@ class WikiTableViewController: UITableViewController {
     fileprivate func createSearchRequest() {
         latestWikiRequest?.cancel()
         guard let searchText = searchText,
-            let wikiRequest = WikiRequest(searchText: searchText) else {
-                return
-        }
-        guard let request = setCompletionBlock(for: wikiRequest) else { return }
-        latestWikiRequest = request
-        request.resume()
-    }
-    
-    fileprivate func setCompletionBlock(for wikiRequest: WikiRequest) -> URLSessionDataTask? {
-        let task = wikiRequest.fetchWikiDocs { [weak self] newDocs in
+            !searchText.isEmpty,
+            let wikiRequest = WikiRequest(searchText: searchText, pageIndex: pageIndex)
+            else { return }
+        latestWikiRequest = wikiRequest
+        pendingSearch = true
+        setCompletionBlock(for: wikiRequest)
+        wikiRequest.task?.resume()
+    }
+    
+    fileprivate func setCompletionBlock(for wikiRequest: WikiRequest) {
+        let isFirstResult = (wikiRequest.offset == 0)
+        wikiRequest.fetchWikiDocs { [weak self] newDocs in
             guard let strongSelf = self else { return }
             DispatchQueue.main.async(){
+                if wikiRequest.isCancelled {
+                    print("Cancelled in search completion block")
+                    return
+                }
                 if newDocs.count > 0 {
-                    strongSelf.imageTasks.cancelAllOperations() // Clears the OperationQueue with consecutive searches
-                    strongSelf.wikiDocs = [newDocs]
+                    strongSelf.clearPendingImageTasks() // Remove stale tasks created by consecutive searches
+                    strongSelf.pendingSearch = false
+                    if isFirstResult {
+                        strongSelf.wikiDocs = [newDocs]
+                    } else {
+                        strongSelf.wikiDocs[0] += (newDocs)
+                    }
                     strongSelf.tableView.reloadData()
```

Comment:
If it's not the first result we probably don't want to reload the whole tableView. Look into the APIs for adding individual tableView rows dynamically, i.e. `func insertRows(at indexPaths: [IndexPath], with animation: UITableViewRowAnimation)`